src/rdkafka_cgrp.c

```diff
@@ -3785,13 +3815,17 @@ rd_kafka_error_t *rd_kafka_consumer_group_metadata_write (
         size_t of = 0;
         size_t magic_len = sizeof(rd_kafka_consumer_group_metadata_magic);
         size_t groupid_len = strlen(cgmd->group_id) + 1;
+        size_t generationid_len = sizeof(int32_t);
 
-        size = magic_len + groupid_len;
+        size = magic_len + groupid_len + generationid_len;
         buf = rd_malloc(size);
 
         memcpy(buf, rd_kafka_consumer_group_metadata_magic, magic_len);
         of += magic_len;
 
+        *(int32_t *)(buf+of) = htole32(cgmd->generation_id);
```

Comment:
Unaligned casts are not allowed, do a memcpy instead.

Reply:
Not sure we need the endian swap since the serialized metadata needs to be used with the same client instance anyway.

Reply:
memcpy: yeah, I shouldn't really need to be told this...
endian swap: yeah, it didn't work initially and i went overboard amending it.